go-controller/pkg/node/gateway_localnet.go

```diff
@@ -52,19 +52,29 @@ func newLocalGateway(nodeName string, hostSubnets []*net.IPNet, gwNextHops []net
 
 	if exGwBridge != nil {
 		gw.readyFunc = func() (bool, error) {
-			ready, err := gatewayReady(gwBridge.patchPort)
-			if err != nil {
-				return false, err
+			for _, netConfig := range gwBridge.netConfig {
```

Comment:
its a good question, but I think we are OK here because we access this only from 1 spot in code... before making this change it was gwBridge.patchPort which was accessed without a lock...
@dceara do you agree?

Reply:
It does look weird to access gwBridge internals without the lock.  OTOH, we just created this bwBridge, 8 lines above, there's nobody else that can access it.

I'm not sure though what would look best from a golang perspective..  Maybe a way to pass a callback to a new `bridgeConfiguration` method that iterates on `bridgeConfiguration.netConfig` and calls the callback for each entry?  A sort of [fold function](https://en.wikipedia.org/wiki/Fold_(higher-order_function) for `bridgeConfiguration`.   That could take the lock internally.

Reply:
ack makes sense I guess better safe than sorry => +1 to confusing part and I didn't realize nodeIP Manager also edited stuff on bridge configuration..

Reply:
done! added locks here